packages/fuels/tests/bindings.rs

```diff
@@ -926,3 +926,42 @@ async fn shared_types() -> Result<()> {
 
     Ok(())
 }
+
+#[cfg(feature = "test-type-paths")]
+#[tokio::test]
+async fn type_paths_respected() -> Result<()> {
+    setup_contract_test!(
+        Wallets("wallet"),
+        Abigen(
+            name = "ContractA",
+            abi = "packages/fuels/tests/bindings/type_paths"
+        ),
+        Deploy(
+            name = "contract_a_instance",
+            contract = "ContractA",
+            wallet = "wallet"
+        ),
+    );
+
+    {
+        let contract_a_type =
+            abigen_bindings::contract_a_mod::contract_a_types::VeryCommonNameStruct {
```

Comment:
Ideally, and arguably most often, the users' structs will have unique names across all the participants of the `abigen!` call. As such, they will be reexported (`pub use`-ed) and the user will refer to the types by just their names.

Less ideal is the example above where `VeryCommonNameStruct` appears multiple times prohibiting its reexporting. The user must then read the `abigen` titled docs section where this is explained.

There is also a section about enabling forc type paths and how the generated type path will need to look in case the user actually needs to fully qualify the type.